frontend/src/reducer/notificationPage.js

```diff
@@ -49,42 +61,220 @@ export const serializeNotification = notification => {
   }
 }
 
+const hasSameAuthor = authorList => {
+  if (authorList.some(author => !author)) return false
+  return authorList.every((author, index) => {
+    if (index === 0) return true
+    return author.userId === authorList[index - 1].userId
+  })
+}
+
+const hasSameWorkspace = workspaceList => {
+  if (workspaceList.some(workspace => !workspace)) return false
+  return workspaceList.every((workspace, index) => {
+    if (index === 0) return true
+    return workspace.id === workspaceList[index - 1].id
+  })
+}
```

Comment:
Same suggestion as for authors.

Using some allows grouping your checks and stopping as soon as something don't match.
(I haven't tried my suggestion)

Reply:
will fix